src/WebJobs.Script.WebHost/Middleware/HttpThrottleMiddleware.cs

```diff
@@ -28,9 +29,11 @@ public HttpThrottleMiddleware(RequestDelegate next, ILoggerFactory loggerFactory
             _logger = loggerFactory?.CreateLogger("Host." + ScriptConstants.TraceSourceHttpThrottleMiddleware);
         }
 
-        public static bool ShouldEnable(HttpOptions options)
+        public static bool ShouldEnable(IScriptHostManager scriptHostManager)
         {
-            return HttpRequestQueue.IsEnabled(options) || options.DynamicThrottlesEnabled;
+            var options = (scriptHostManager as IServiceProvider)?.GetService<IOptions<HttpOptions>>();
```

Comment:
I believe there's a chance you'll hit a race and get an `ObjectDisposedException` here if the host is stopping or restarting. I know elsewhere we try/catch for this explicitly. I think we'll want to do this here too.

Reply:
Added an extension method and used it in a few places where we're already using this pattern

Reply:
Seeing some test failures - investigating